src/build/__init__.py

```diff
@@ -187,57 +187,51 @@ def _parse_build_system_table(pyproject_toml: Mapping[str, Any]) -> Dict[str, An
 
 
 class ProjectBuilder:
-    """
-    The PEP 517 consumer API.
-    """
+    """The PEP 517 consumer API."""
 
-    _default_rewrapped_runner = staticmethod(rewrap_runner_for_pep517_lib(default_runner))
+    @overload
+    def __init__(self, srcdir: PathType, *, python_executable: str = ..., runner: RunnerType = ...) -> None:
+        ...
+
+    @overload
+    def __init__(self, srcdir: PathType, *, isolated_env: env.IsolatedEnv, runner: RunnerType = ...) -> None:
+        ...
 
     def __init__(
         self,
         srcdir: PathType,
+        *,
         python_executable: str = sys.executable,
-        runner: Optional[Union[RunnerType, Tuple[RunnerType, Optional[Mapping[str, str]]]]] = None,
+        isolated_env: Optional[env.IsolatedEnv] = None,
+        runner: RunnerType = default_runner,
     ) -> None:
         """
-        :param srcdir: The project source directory
-        :param python_executable: Path of Python executable used to invoke
-            PEP 517 hooks
+        :param srcdir: Project source directory
+        :param python_executable: Path of Python executable used to invoke PEP 517 hooks
+        :param isolated_env: An :class:``env.IsolatedEnv`` instance
         :param runner: Callback for executing PEP 517 hooks in a subprocess
         """
         self._srcdir = _parse_source_dir(srcdir)
-        self._python_executable = python_executable
+
+        if isolated_env is not None:
```

Comment:
Could use `**kwargs` and raise `TypeError` if `isolated_env` is passed along with `python_executable` but I thought it'd be overkill.